tests/testthat/test-pws.R

```diff
@@ -0,0 +1,307 @@
+  app$set_inputs(`plot_with_settings-width_in_modal` = 500)
+  app$set_inputs(`plot_with_settings-modal_downbutton-file_name` = "plot_pdf")
+  app$set_inputs(`plot_with_settings-modal_downbutton-file_format` = "pdf")
+  app$set_inputs(`plot_with_settings-height_in_modal` = 1200)
+  app$set_inputs(`plot_with_settings-width_in_modal` = 750)
+  app$click("plot_with_settings-modal_downbutton-downl")
+  app$set_inputs(`plot_with_settings-modal_downbutton-downl_state` = TRUE)
+
+  vals <- app$get_values()
+  testthat::expect_equal(
+    shiny::isolate(vals$input$`plot_with_settings-modal_downbutton-file_name`),
+    "plot_pdf"
+  )
+  testthat::expect_equal(
+    shiny::isolate(vals$input$`plot_with_settings-height_in_modal`), 1200
+  )
+  testthat::expect_equal(
+    shiny::isolate(vals$input$`plot_with_settings-width_in_modal`), 750
+  )
+
+  # nolint start
+  testthat::expect_equal(
+    shiny::isolate(vals$output$`plot_with_settings-plot_modal`$width), 750
+  )
+  testthat::expect_equal(
+    shiny::isolate(vals$output$`plot_with_settings-plot_modal`$height), 1200
+  )
+  # nolint end
+
+  # change to svg
+  app$set_inputs(`plot_with_settings-modal_downbutton-downl_state` = TRUE)
+  app$set_inputs(`plot_with_settings-modal_downbutton-file_format` = "svg")
+  app$set_inputs(
+    `plot_with_settings-modal_downbutton-file_name` = "plot_svg_modal"
+  )
+  testthat::expect_equal(
+    app$get_value(input = "plot_with_settings-modal_downbutton-file_name"),
+    "plot_svg_modal"
+  )
+
+  # reset
+  app$set_inputs(`plot_with_settings-modal_downbutton-downl_state` = FALSE)
+})
+
+# Testing hide and show button
+testthat::test_that("plot_with_settings: hide/show button", {
+  skip_on_cran()
+  skip_on_ci()
+
+  # nolint start
+  # visible on load
+  testthat::expect_true(
+    app$get_js(
+      "$('#plot_with_settings-plot-with-settings').is(':visible')"
+    )
+  )
+
+  # hide
+  app$click("button")
+  testthat::expect_true(
+    app$get_js(
+      "$('#plot_with_settings-plot-with-settings').is(':hidden')"
+    )
+  )
+
+  # unhide
+  app$click("button")
+  testthat::expect_true(
+    app$get_js(
+      "$('#plot_with_settings-plot-with-settings').is(':visible')"
+    )
+  )
+  # nolint end
+})
+
+# tests width warning displays when width too low, hides when not.
+# note that warning is not hidden/visible in the usual sense.
+# rather, it has the fa icon <span> as a child or it does not.
+# hence we're checking number of children.
+testthat::test_that("plot_with_settings: width warning", {
+  skip_on_cran()
+  skip_on_ci()
+
+  app$click("plot_with_settings-expbut")
+
+  # starts out visible
+  testthat::expect_equal(
+    app$get_js("$('#plot_with_settings-width_warning').children().length"), 1
+  )
+  testthat::expect_equal(
+    app$get_value(output = "plot_with_settings-width_warning")$html[1],
+    paste0(
+      "<span class=\"help-block\">\n  <i class=\"fa fa-triangle-exclamation\" role=\"presentation\" aria-label=",
```

Comment:
I will remove this tests. Locally, and in cicd, fa changed fo fas `fas fa-triangle-exclamation`, so it is unstable and let's skip it.